addons/pos_restaurant/models/pos_restaurant.py

```diff
@@ -30,6 +31,17 @@ def unlink(self):
                 raise UserError(error_msg)
         return super(RestaurantFloor, self).unlink()
 
+    def write(self, vals):
+        open_session_ids = self.env['pos.session'].search([('config_id', 'in', self.pos_config_id.ids), ('state', '!=', 'closed')])
+        is_used = self.env['pos.config'].search([('floor_ids', '=', self.id)])
```

Comment:
the function write is multi so you need a loop, for floor in self.

Then, there is only one pos config per floor, so no need to search it, and we can easily know if it has a session open.

So here is how you could simplify it:
- do the loop
- if floor.pos_config_id.has_active_session -> raise open in config blabla
- if vals.get(pos_config_id) and vals.get(pos_config_id) != floor.pos_config_id -> raise already used in another config

Reply:
The call to super can be out of the loop used with self